code/modules/virus2/items_devices.dm

```diff
@@ -1,74 +1,389 @@
 	icon_state = "antibody"
 	w_class = W_CLASS_SMALL
 	item_state = "electronic"
 	flags = FPRINT
 	siemens_coefficient = 1
 
 
-/obj/item/device/antibody_scanner/attack(mob/living/carbon/M as mob, mob/user as mob)
-	if(!istype(M))
-		to_chat(user, "<span class='notice'>Incompatible object, scan aborted.</span>")
+/obj/item/device/antibody_scanner/attack(var/mob/living/L, var/mob/user)
+	if(!istype(L))
+		//to_chat(user, "<span class='notice'>Incompatible object, scan aborted.</span>")
 		return
-	var/mob/living/carbon/C = M
-	if(!C.antibodies)
-		to_chat(user, "<span class='notice'>Unable to detect antibodies.</span>")
+
+	playsound(user, 'sound/items/detscan.ogg', 50, 1)
+	var/info = ""
+	var/icon/scan = icon('icons/virology.dmi',"immunitybg")
+	if (L.immune_system)
+		var/immune_system = L.immune_system.GetImmunity()
+		var/immune_str = immune_system[1]
+		var/list/antibodies = immune_system[2]
+
+		info += "Immune System Status: <b>[round(immune_str*100)]%</b>"
+		info += "<br>Antibody Concentrations:"
+
+		var/i = 0
+		for (var/antibody in L.immune_system.antibodies)
+			var/rgb = "#80DEFF"
+			switch (i)
+				if (4 to 6)
+					rgb = "#81FF9F"
+				if (7 to 9)
+					rgb = "#E6FF81"
+				if (10 to 12)
+					rgb = "#FF9681"
+			scan.DrawBox(rgb,i*43+11,6,i*43+31,6+antibodies[antibody]*3*immune_str)
+			i++
+
+	if (L.virus2.len)
+		for (var/ID in L.virus2)
+			var/datum/disease2/disease/D = L.virus2[ID]
+			scan.DrawBox("#FF0000",5,6+D.strength*3,564,6+D.strength*3)
+			if(ID in virusDB)
+				var/subdivision = (D.strength - ((D.robustness * D.strength) / 100)) / D.max_stage
+				var/i = 0
+				for (var/antigen in all_antigens)
+					if (antigen in D.antigen)
+						scan.DrawBox("#FF0000",18+43*i,6+D.strength*3-3,24+43*i,6+D.strength*3+3)
+						scan.DrawBox("#FF0000",21+43*i,6+D.strength*3,21+43*i,6+round(D.strength - D.max_stage * subdivision)*3)
+						for (var/j = 1 to D.max_stage)
+							var/alt = round(D.strength - j * subdivision)
+							scan.DrawBox("#FF0000",5+43*i,6+alt*3,38+43*i,6+alt*3)
+					i++
+
+	info += "<br><img src='data:image/png;base64,[icon2base64(scan)]'/>"
+	info += "<br>"
+	info += "<table style='table-layout:fixed;width:560px;text-align:center'>"
+	info += "<tr>"
+	if (L.immune_system)
+		for (var/antibody in L.immune_system.antibodies)
+			info += "<th>[antibody]</th>"
+	info += "</tr>"
+	info += "<tr>"
+	if (L.immune_system)
+		for (var/antibody in L.immune_system.antibodies)
+			info += "<td>[round(L.immune_system.antibodies[antibody]*L.immune_system.strength)]%</th>"
+	info += "</tr>"
+	info += "</table>"
+
+	if (L.virus2.len)
+		for (var/ID in L.virus2)
+			var/datum/disease2/disease/D = L.virus2[ID]
+			if(ID in virusDB)
+				var/datum/data/record/V = virusDB[ID]
+				info += "<br><i>[V.fields["name"]][V.fields["nickname"] ? " \"[V.fields["nickname"]]\"" : ""] detected. Strength: [D.strength]. Robustness: [D.robustness]. Antigen: [D.get_antigen_string()]</i>"
```

Comment:
As an aside, this line seems to break syntax highlighting. Don't know why.

Reply:
:woman_shrugging: